### ISSUE TITLE
Routing config leads to two `Prefix /` lines in Routing view

### ISSUE THREAD

**Issue by lkysow** [hashicorp/consul#11874]
#### Overview of the Issue

Consul 1.11.1

![image](https://user-images.githubusercontent.com/1034429/146460222-81de2c2c-53d5-4037-aaa9-f32ef20b4aae.png)

```hcl
Kind   = "service-router"
Name   = "backend"
Routes = [
  {
    Destination = {
      RequestTimeout     = "1s"
    }
  }
]
```

In this case since the two boxes are the same "Prefix /" I'd only expect one box since this doesn't help with my routing understanding.

**Comment by johncowen** [comment]
Hey @lkysow 

I had a look at the HTTP response from Consul for this and it looks like:

<img width="660" alt="Screenshot 2022-01-10 at 11 31 46" src="https://user-images.githubusercontent.com/554604/148759389-23ae4c78-75a0-43e4-99a2-bbb5479ee226.png">

Note that there are 2 Routes in the response. Therefore the the UI shows two routes.

I vaguely remember quite a while back that we've had a quite a few conversations amongst the team about a 'default' route and a `/` Route and I think this is the same conversation or at least similar. (From what I remember we chose to show "Prefix /" instead of `Default` or `*` for when a route does not have a PathPrefix defined, which is the case here.

Agree it looks strange in this case, so maybe its worth revisiting to either change or a least document the reasons around why it is visually presented like this.

If one of these was `No Prefix` or maybe `*` would that make more sense from your perspective? (or do you have another suggestion?)

**Comment by lkysow** [comment]
Ahh I think this should be a change in the backend response the potentially. According to the docs on `Match{}` (https://www.consul.io/docs/connect/config-entries/service-router#match)

> A set of criteria that can match incoming L7 requests. If empty or omitted it acts as a catch-all.

In this case it's omitted so it's the catch-all which means that there shouldn't be two routes returned, it should only return the 1.

**Comment by johncowen** [comment]
@rboyer any insight from you side here?

**Comment by lkysow** [comment]
@johncowen another possible thing to do here would be to de-duplicate routes that will be shown as identical in the UI.

**Comment by johncowen** [comment]
Yeah its definitely an option and I see where you are coming from. 'Generally' I'm against showing things differently in the UI to what is actually produced by the backend/API though. Theoretically 2 Routes do actually exist here.

Another thing that could maybe help is to show the `RequestTimeout` value for the Route that has it (we don't currently show a lot of the data that we could show that the endpoint gives us here). Right now it looks strange partly because the 2 Routes look the same in the UI when actually they are different.

**Comment by thomashashi** [comment]
I'm running into this same issue while helping an opportunity, and it confused me to no end. This is particularly confusing from a UX standpoint: no service-router entry, no routes show up in the UI (or when you call `/v1/discovery-chain`); add a service-router entry with a single destination, *two* routes show up. And, I can't distinguish them in the UI, so now I'm wondering what I did wrong to end up with two entries.

Now, I happen to know about `/v1/discovery-chain`, so I can "look at what's really going on under the covers", but then I see my entry and some I had nothing to do with. My first thought was "maybe Consul added an ultimate default" automatically but then I was "but the documentation says a destination entry with no match matches everything and acts as a catch-all". I *think* I can convince myself that my entry, as the first one, will catch things, but I've got underlying confusion and I'm not totally certain.

1. Is that second entry the result of "if there is any route added we put an 'internal default' at the end"?
2. If that's the case, if something which is logically a "catch-all" at the end of the 'user-supplied' destination array, a) can we identify a user-added entry which *is* logically a catch-all?; b) if so, can we not add the 'internal default'?
3. If we do need to keep the 'internal default', can we mark it as such in `/v1/discovery-chain` and in the UI?
4. How can we document this to reduce customer confusion?
5. How can someone who sees this 'internal default' which looks like it comes after their added "catch-all" entry convince themself that requests are using the entry they added, with the options they set?

**Comment by thomashashi** [comment]
Thanks @johncowen, if there's anything I can do to help let me know — I do nothing but sell Consul and think about Consul and _I_ got confused by this, so I can imagine it would be even more confusing for someone who is simply trying to use Consul.
